docs/proposals/thin-volumes.md

```diff
@@ -0,0 +1,175 @@
+``` yaml=
+device-classes:
+  - name: ssd-thin
+    volume-group: myvg1
+    default: true
+    spare-gb: 10
+    type: thin
+    thin-pool:
+        name: pool0
+        overprovision-ratio: 50.0
+  - name: ssd
+    volume-group: myvg1
+    default: true
+    spare-gb: 10
+```
+- Above example shows two device classes; `ssd-thin` and `ssd`. Both use `myvg1` volume group.
+- `ssd-thin` has thin pool parameters.
+- Only thin volumes will be created when `ssd-thin` device class is selected by the user in the storageClass parameter `capacity.topolvm.cybozu.com/device-class`
+- Only thick volumes will be created when user has specified `ssd` in the storageClass.
+
+#### API Changes
+
+- The `DeviceClass` struct of the device-class-manager will hold the information about the thin-pool associated with the Volume Group.
+
+``` go
+
++ type ThinPoolConfig struct {
++   // Name of thin pool
++   Name string`json:"name"`
++   // OverProvisionRatio represents the ratio of overprovision that can be + allowed on thin pools
++   OverProvisionRatio float64 `json:"overprovision-ratio"`
++}
+
+type DeviceClass struct {
+    // Name for the device-class name
+    Name string `json:"name"`
+    // Volume group name for the device-class
+    VolumeGroup string `json:"volume-group"`
+    // Default is a flag to indicate whether the device-class is the default
+    Default bool `json:"default"`
+    // SpareGB is storage capacity in GiB to be spared
+    SpareGB *uint64 `json:"spare-gb"`
+    // Stripe is the number of stripes in the logical volume
+    Stripe *uint `json:"stripe"`
+    // StripeSize is the amount of data that is written to one device before moving to the next device
+    StripeSize string `json:"stripe-size"`
+    // LVCreateOptions are extra arguments to pass to lvcreate
+    LVCreateOptions []string `json:"lvcreate-options"`
++   // Type of the volume to be created
++   Type string `json:"type",omitempty`
++   // ThinPoolConfig represents the configuration options for creating thin volumes
++   ThinPoolConfig ThinPoolConfig `json:"thin-pool",omitempty`
+}
+```
+- Following new fields will be added to the API:
+    - **Type**: Represents the type of the volume provisioning. Possible values are `thin` and `thick`. This field is optional. If no type is provided, then thick volumes should be provisioned.
+    - **ThinPoolConfig**: Shows the configuration of thin pool used for creating thin volumes. This field must exist if the `type` field is `thin` and ignored if the `type` field is `thick`.
+        - **Name**: Name of the thin pool
+        - **OverProvisionRatio**: This represents the amount of overprovisioning allowed for the pool. For example: If the overprovisioningRatio is `5`, pool size is 100GiB, then this pool can be overprovisioned by only (5*100)500GiB.
+
+
+#### Selecting Thin-pool
+- The current behavior uses `topolvm.cybozu.com/device-class` parameter on the deviceClass to decide which volume group will be used.
```

Comment:
Please clarify this  - the annotation is not set on the deviceClass.

Reply:
Right.  Its on the StorageClass.  Updated.